src/applications/apps.js

```diff
@@ -26,7 +26,7 @@ export function declareChildApplication(appName, arg1, arg2) {
   return registerApplication(appName, arg1, arg2)
 }
 
-export function registerApplication(appName, arg1, arg2) {
+export function registerApplication(appName, arg1, arg2, customProps) {
```

Comment:
Nit: you could make customProps default to empty object here and avoid [this if statement](https://github.com/CanopyTax/single-spa/pull/156/files#diff-09ec3011a4b92de4d61e29380e950bc3R60) and [this if statement](https://github.com/CanopyTax/single-spa/pull/156/files#diff-b954ca9fc4221dce0e4e9cffc6a642c9R64) later.

```js
export function registerApplication(appName, arg1, arg2, customProps = {}) {
  ...
}
```

Reply:
Great idea.